src/main/scala/org/apache/sysml/api/ml/BaseSystemMLClassifier.scala

```diff
@@ -65,8 +67,25 @@ trait HasRegParam extends Params {
   final def getRegParam: Double = $(regParam)
 }
 
-trait BaseSystemMLEstimator {
-  
+trait BaseSystemMLEstimatorOrModel {
+  var enableGPU:Boolean = false
+  var explain:Boolean = false
+  var statistics:Boolean = false
+  val config:HashMap[String, String] = new HashMap[String, String]()
+  def setGPU(enableGPU1:Boolean):BaseSystemMLEstimatorOrModel = { enableGPU = enableGPU1; this}
+  def setExplain(explain1:Boolean):BaseSystemMLEstimatorOrModel = { explain = explain1; this}
+  def setStatistics(statistics1:Boolean):BaseSystemMLEstimatorOrModel = { statistics = statistics1; this}
+  def setConfigProperty(key:String, value:String):BaseSystemMLEstimatorOrModel = { config.put(key, value); this}
+  def updateML(ml:MLContext):Unit = {
```

Comment:
Can you change this to `def createML(sc: SparkContext): MLContext = {....}` to avoid the pattern of relying on side-effects?  The side effects style is prone to subtle bugs.  All uses below can just pass a `SparkContext` object to this function and receive a new `MLContext` object, rather than create an `MLContext` object and then ask for it to be updated.